lib/lib.go

```diff
@@ -0,0 +1,229 @@
+package lib
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"sync"
+
+	"github.com/spf13/viper"
+)
+
+// Khan is a struct that represents a khan API application
+type Khan struct {
+	httpClient *http.Client
+	Config     *viper.Viper
+	url        string
+	user       string
+	pass       string
+	gameID     string
+}
+
+var (
+	client *http.Client
+	once   sync.Once
+)
+
+func getHTTPClient() *http.Client {
+	once.Do(func() {
+		client = &http.Client{}
```

Comment:
Regarding Jaeger, you did not instrument the http calls. Also, I'd recommend setting a default timeout for the requests and probably configuring keepalive (I'm not sure which are the default values).

```
import ehttp "github.com/topfreegames/extensions/http"
client = &http.Client{
	Timeout: 1 * time.Second, // or some value from the config
}

ehttp.Instrument(client)
```